src/loan/LendingTerm.sol

```diff
@@ -170,11 +170,63 @@ contract LendingTerm is CoreRef {
         // initialize storage
         _setCore(_core);
         refs = _refs;
-        params = _params;
+        params = abi.decode(_params, (LendingTermParams));
+
+        // check parameters:
+        // must be an ERC20 (maybe, at least it prevents dumb input mistakes)
+        (bool success, bytes memory returned) = params.collateralToken.call(
+            abi.encodeWithSelector(IERC20.totalSupply.selector)
+        );
+        require(
+            success && returned.length == 32,
+            "LendingTerm: invalid collateralToken"
+        );
+
+        require(
+            params.maxDebtPerCollateralToken != 0, // must be able to mint non-zero debt
+            "LendingTerm: invalid maxDebtPerCollateralToken"
+        );
+
+        require(
+            params.interestRate < 1e18, // interest rate [0, 100[% APR
+            "LendingTerm: invalid interestRate"
+        );
+
+        require(
+            params.maxDelayBetweenPartialRepay < YEAR + 1, // periodic payment every [0, 1 year]
+            "LendingTerm: invalid maxDelayBetweenPartialRepay"
+        );
+
+        require(
+            params.minPartialRepayPercent < 1e18, // periodic payment sizes [0, 100[%
+            "LendingTerm: invalid minPartialRepayPercent"
+        );
+
+        require(
+            params.openingFee <= 0.1e18, // open fee expected [0, 10]%
+            "LendingTerm: invalid openingFee"
+        );
+
+        require(
+            params.hardCap != 0, // non-zero hardcap
+            "LendingTerm: invalid hardCap"
+        );
+
+        // if one of the periodic payment parameter is used, both must be used
+        if (
+            params.minPartialRepayPercent != 0 ||
+            params.maxDelayBetweenPartialRepay != 0
+        ) {
+            require(
+                params.minPartialRepayPercent != 0 &&
+                    params.maxDelayBetweenPartialRepay != 0,
+                "LendingTerm: invalid periodic payment params"
+            );
+        }
```

Comment:
Now that the parameters are correctly decoded, we can test the assumptions

Reply:
| Contract                   | Size (kB) | Margin (kB) |
|----------------------------|-----------|-------------|
| LendingTerm                | 18.714    | 5.862       |

LGTM